--- a/python/setup.py.in	2020-02-20 20:04:33.000000000 +0000
+++ b/python/setup.py.in	2020-02-20 20:04:57.000000000 +0000
@@ -13,7 +13,7 @@
                                     "${CMAKE_CURRENT_COURCE_DIR}/rpc.h"
                                    ],
                            libraries=["netconf2"],
-                           extra_compile_args=["-Wall", "-I${CMAKE_CURRENT_BINARY_DIR}" @SSH_DEFINE@ @TLS_DEFINE@],
+                           extra_compile_args=["-Wall", "-I${CMAKE_CURRENT_BINARY_DIR}", "-I${LIBYANG_INCLUDE_DIR}", "-I${LIBSSH_INCLUDE_DIR}" @SSH_DEFINE@ @TLS_DEFINE@],
                            extra_link_args=["-L${CMAKE_CURRENT_BINARY_DIR}/.."],
                         )

